### PR TITLE
Make Ids work in UpdateCOmmand

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

**Comment by pranavkm** [comment]
cc @deepakaravindr \ @emgarten

**Comment by pranavkm** [comment]
Don't think this change warrants being performant as much as it is for back compat. It was incredibly slow in the past, so we're not starting off any worse.

> exceptions are thrown due to no-op updates

That sounds like a bug in PackageMangement. I can have a look at the how we handle dependency conflicts.

**Comment by deepakaravindr** [comment]
I don't think exceptions are thrown due to no-op updates. They get thrown if a package is already installed or if a newer version is already installed and so on. Good to double check that

For the -Id switch case alone and the current milestone, I would not worry about performance either

File: src/NuGet.CommandLine/Commands/UpdateCommand.cs [hunk 1]
```diff
@@ -228,20 +228,44 @@ private Task UpdatePackagesAsync(string packagesConfigPath, INuGetProjectContext
             var projectDirectory = Path.GetDirectoryName(project.ProjectFullPath);
             var packageManager = new NuGetPackageManager(sourceRepositoryProvider, Settings, packagesDirectory);
             var nugetProject = new MSBuildNuGetProject(project, packagesDirectory, projectDirectory);
+            var versionConstraints = Safe ? VersionConstraints.ExactMajor | VersionConstraints.ExactMinor : VersionConstraints.None;
```
Comment (deepakaravindr) [inline src/NuGet.CommandLine/Commands/UpdateCommand.cs:231]: break the line

Comment (deepakaravindr) [inline src/NuGet.CommandLine/Commands/UpdateCommand.cs:231]: Why don't you do a 'var resolutionContext' and use that in the 2 calls to PreviewUpdatePackagesAsync. That will make the code even more readable

**Comment by deepakaravindr** [comment]
:shipit:. Before check-in, I would quickly check if the negative cases, such as package already installed or newer version is already available, work like before

**Pull request closed without merging**